capture/crawler_estadao.py

```diff
@@ -0,0 +1,189 @@
+             }
+    try:
+        date = soup.findAll("p", {"class":"data"})[0]
+    except IndexError:
+        try:
+            date = soup.findAll("span", {"class":"data"})[0]
+        except IndexError:
+            logger.error('wrong date tags')
+            return None
+
+    try:
+        date = date.text.strip().split()
+        date[1] = date[1].lower()
+        date[1] = MONTHS[date[1]]
+
+        if "estadao.com.br/noticias/" in url:
+            date = date[0:3] + date[4:6]
+            date[3] = date[3][0:2]
+
+        elif "estadao.com.br/blogs/" in url:
+            date[3] = date[4][0:2]
+            date[4] = date[4][3:5]
+    except ValueError:
+        logger.error('wrong data extraction')
+        return None
+
+    date = '-'.join(date)
+
+    try:
+        published_time = datetime.datetime.strptime(date, '%d-%b-%Y-%H-%M')
+    except ValueError:
+        logger.error('wrong published time format')
+        return None
+    return published_time
+
+def extract_title(article):
+    """ Extract the news title.
+    """
+
+    try:
+        title = article.title
+    except Exception as ex:
+        template = "An exception of type {0} occured during extraction of news title. Arguments:\n{1!r}"
+        message = template.format(type(ex).__name__, ex.args)
+        logger.exception(message)
+        return None
+    return title
+
+def extract_content(article):
+    """ Extract relevant information about news page
+    """
+
+    try:
+        body_content = article.cleaned_text
+    except Exception as ex:
+        template = "An exception of type {0} occured during extraction of news content. Arguments:\n{1!r}"
+        message = template.format(type(ex).__name__, ex.args)
+        logger.exception(message)
+        return None
+    return body_content
+
+def download_article(url):
+    """ Download the html content of a news page
+
+    :param url: news page's url
+    :type url: string
+    :return: news page's content
+    :rtype: requests.models.Response
+    """
+
+    article = { 'link': url, 'source': 'crawler_estadao' }
+    logger.info("Downloading article: {0}".format(url))
+
+    try:
+        response = requests.get(url, timeout=30)
+    except ConnectionError:
+        logger.error("Failed to fetch:{0}".format(url))
+        return None
+    except Timeout:
+        logger.error("Timed out while fetching {0}".format(url))
+        return None
+
+    encoding = response.encoding if response.encoding is not None else 'utf8'
+    response_content = response.content.decode(encoding)
+    soup = BeautifulSoup(response_content)
+
+    extractor = Goose({'use_meta_language': False, 'target_language':'pt'})
+    news = extractor.extract(url=url)
+
+    article['link_content'] = compress_content(response_content)
+    article['compressed'] = True
+    article['title'] = extract_title(news)
+    article['body_content'] = extract_content(news)
+    article['published_time'] = extract_published_time(url, soup)
```

Comment:
`extract_title`,  `extract_content` and `extract_published_time` will also sometimes return `None`. In this case, it will mean the document will have `null` in the database as a value for this attribute. 

I think this may be what we want (better to have the content and no title than to have nothing at all), but I am not 100% sure.

@fccoelho any ideas on this?